src/server.cpp

```diff
@@ -2172,6 +2172,17 @@ inline s32 Server::nextSoundId()
 	return free_id;
 }
 
+std::pair<
+	std::unordered_map<s32, ServerPlayingSound>::iterator,
+	std::unordered_multimap<u16, s32>::iterator
+> Server::removeSoundReference(std::unordered_map<s32, ServerPlayingSound>::iterator it)
+{
+	auto range = m_attached_sound_ids.equal_range(it->second.object);
```

Comment:
(this should be empty if `it->second.object` is `0`)

Reply:
(if you have to mention this it might be best to add it as a comment to the code)

Also: *should* ? You could either add an `assert` or separate condition to ensure that. `m_attached_sound_ids` should never have object ID 0.

Reply:
> `m_attached_sound_ids` should never have object ID 0.

Yes, hence why this range should be empty in that case.